src/app/zones/data/index.ts

```diff
@@ -1,36 +1,87 @@
+  zoneInsight?: ZoneInsight
+  zone: Zone
+  state: 'online' | 'offline' | 'disabled'
 }
-export function getZoneDpServerAuthType(zone: ZoneOverview): string {
-  const subscriptions = zone.zoneInsight?.subscriptions ?? []
-  if (subscriptions.length > 0) {
-    const lastSubscription = subscriptions[subscriptions.length - 1]
-    if (lastSubscription.config) {
-      const parsedConfig = JSON.parse(lastSubscription.config)
-      return get(parsedConfig, 'dpServer.auth.type', '')
+
+export const Zone = {
+  fromObject: (item: PartialZone): Zone => {
+    return {
+      ...item,
+      enabled: !(item.enabled === false),
     }
-  }
-  return ''
+  },
 }
-// TODO(jc): end
+export const ZoneInsight = {
+  fromObject: (item?: PartialZoneInsight): ZoneInsight | undefined => {
+    // if item isn't set don't even try augmenting things
+    return isSet<PartialZoneInsight>(item)
+      ? ((item) => {
+        // figure out the connectedSubscription by looking at the connectTime
+        // and disconnectTime of the last subscription
+        const connectedSubscription: KDSSubscription | undefined = !Array.isArray(item.subscriptions)
+          ? undefined
+          : [item.subscriptions[item.subscriptions.length - 1]].find((item) => item.connectTime?.length && !item.disconnectTime)
+
+        // using the connectedSubscription find the config for the zone if it exists, is valid JSON and is not null and
+        // turn it into an object
+        const config: Record<string, unknown> = (() => {
+          const str = isSet<string>(connectedSubscription?.config) ? connectedSubscription.config : '{}'
+          try {
+            return JSON.parse(str)
+          } catch (e) { }
+          return {}
+        })()
 
-// The presence of a `ZoneOverview.zoneInsight` object's subscriptions
-// with a connect time and without a disconnect time indicate a Zone to
-// be connected and online.
-export function getZoneControlPlaneStatus(zoneOverview: ZoneOverview): 'online' | 'offline' | 'disabled' {
-  if (zoneOverview.zone.enabled === false) {
-    return 'disabled'
-  }
-  return getIsConnected(zoneOverview.zoneInsight?.subscriptions) ? 'online' : 'offline'
+        // set the extra stuff
+        return {
+          ...item,
+          connectedSubscription,
+          config,
+          authenticationType: get(config, 'dpServer.auth.type', ''),
+          environment: String(config.environment ?? ''),
+        }
+      })(item)
+      : undefined
+  },
+}
+export const ZoneOverview = {
+  fromObject: (item: PartialZoneOverview): ZoneOverview => {
+    const insight = ZoneInsight.fromObject(item.zoneInsight)
+    const zone = Zone.fromObject(item.zone)
+    return {
+      ...item,
+      zoneInsight: insight,
+      zone,
+      // first check see if the zone is disabled, if not look for the connectSubscription
+      state: !zone.enabled ? 'disabled' : typeof insight?.connectedSubscription !== 'undefined' ? 'online' : 'offline',
```

Comment:
We use `status` elsewhere for an object name not a string FSM state. This is a string FSM state, not an object of properties/data/infinite state. `state` describes this characteristic very well and would not be confused with the `status` object elsewhere. How big of an issue is this?

Reply:
Okay, if you insist, then let’s call it state. Can you please create an issue to rename this to state everywhere then? We call it status virtually everywhere and introducing a different name here will be confusing over time.

Reply:
Can you be a little more specific here, I'm guessing you don't mean the backend properties, seeing as we can't really change those?

I think as we move forwards adding our 'frontend' properties to our 'backend' types I'll add a `state` on another few entities, but seeing as this is the first instance of us having these 'frontend properties' theres nowhere to change, I'll just make these new `state` props as I go?

I wondered of you meant things like `ResourceStatus` but they can and do show more than just the `state`, so status is kinda fine for those. Anyway lemme know, i can certainly add an issue if I know exactly what we need to change.

I just want to point out again that `state` is meant to represent a single finite state, not a grouping of properties/data.

---

Tiny little note whilst looking at this I think `StatusInfo` isn't used anymore so I'll make a PR to delete that

Reply:
I mean the stuff we virtually everywhere call status. For example, the stuff we pass to StatusBadge, the component, itself, etc.

Reply:
Ok honestly I think `StatusBadge` comes into what I said about `ResourceStatus` but I kinda don't mind either way, feel free to make an issue if you want to go ahead and change those thing I'm not that fussed.

Just to underline again, the new `state` frontend property is specifically to differentiate between the finite state of certain entities and the places where sometimes `status` is a string `state` and sometimes its a bag of properties.

From now on, on all our entities that have a `state`, `state` is the finite state of the thing, and the `status` could be anything.

Reply:
I’m confused about this now. In this PR, you motivated the change to call what we currently call "status" virtually everywhere "state". Personally, I don’t like that because (1) status is already an appropriate name for the concept and (2) state isn’t a better name (it’s more generic).

You want to distinguish a scalar status representation from a representation as an object, but we don’t have any instances where a resource’s status is represented as anything but a simple string. We have Zones (online, offline, disabled), Dataplanes (online, offline, partially degraded), and Services (online, offline, partially degraded, not available). Only in the case of Dataplanes do we accompany the status with a `reason` array to further clarify a certain status. But even in that case, the status is a simple string and not a dictionary.

If we rename this concept in our application, it follows we shouldn’t just do it in some places. That’s why I asked you to create a ticket for this. I don’t want to call this "state" here and then "status" everywhere else. Then we could just keep the name "status" here and it would be less confusing.

---

I previously said I’m fine if you think this is the right thing to do, but I’m not fine with using two different names for the status concept when we currently only have the one which is completely appropriate.

Reply:
What about `status` here?

https://github.com/kumahq/kuma-gui/blob/ec0bb869c7dcd3fdf9241215935c7dfccecfaac2/src/types/index.d.ts#L41

and here

https://github.com/kumahq/kuma-gui/blob/ec0bb869c7dcd3fdf9241215935c7dfccecfaac2/src/types/index.d.ts#L178

?

If you're fine with this, lets stop here. If it ends up being a bad choice we can come back and change.

This convo is resolved so lets leave it at that.